### PR TITLE
#182 fix: Adds paging and sorting to Request requests

### PR DESCRIPTION
**Pull request by dvanwinkle**

### CODE REVIEW

**Comment by mozts2005** [comment]
I would like to see testing. 

Also as these are breaking changes we will need to pair this with a version number change.   

http://stackoverflow.com/questions/1456785/a-definitive-guide-to-api-breaking-changes-in-net 
(see the last answer)

**Comment by dvanwinkle** [comment]
@mozts2005 I will add some testing here as well. Didn't realize that adding default parameters would cause a breaking change, so, good to know! (http://stackoverflow.com/a/23517490/412339 for anyone wondering why.)

I am not familiar with updating version numbers in a NuGet package. I will happily accept a PR from you on my fork/branch, otherwise, can you point me in the right direction?

Thanks!

**Comment by dvanwinkle** [comment]
@mozts2005 Tests are added, however, I still need direction bumping the version numbers.

Thanks!

**Comment by mozts2005** [comment]
@dvanwinkle The version number is located in the appveyor.yml

**Comment by dvanwinkle** [comment]
@mozts2005 I will bump the version number, but, #181 will also need a version number bump. Have you considered using a `develop` branch and then doing formalized releases via something similar to `git-flow`? In that scenario, each PR author wouldn't be responsible for the version number bump, which makes a bit more sense as there is no guarantee when/in what order a PR will be accepted.

File: Tests/RequestTests.cs [hunk 1]
```diff
@@ -9,53 +9,231 @@ namespace Tests
...
+            var res = _api.Requests.GetAllRequests();
             Assert.True(res.Count > 0);
-        }        
+        }
+
+        [TestCase(1, 1)]
+        [TestCase(1, 2)]
+        public void CanGetAllRequestsPaged(int perPage, int page)
+        {
+            Assert.DoesNotThrow(() =>
```
Comment (mozts2005) [inline Tests/RequestTests.cs]: Why do you have Assert.DoesNotThrow?

**Comment by mozts2005** [comment]
I have considered using a develop branch. I have found in other projects that adds a barrier to others joining the project. 

as for the version number I will be managing that going forward. When I mentioned version number earlier it was more to point out that it had to be done not that you need to be the one to update it.

File: Tests/RequestTests.cs (hunk 1 above)
Comment (dvanwinkle) [inline Tests/RequestTests.cs]: `GetAllRequests` can throw an exception. It should not typically throw an exception, but if it does, it should throw an assertion.

**Comment by dvanwinkle** [comment]
@mozts2005 Sorry about the version number confusion. Would you like me to revert that?

**Comment by mozts2005** [comment]
No you don't need to revert the version number

File: Tests/RequestTests.cs (hunk 1 above)
Comment (mozts2005) [inline Tests/RequestTests.cs]: I can see what you are saying, but note the test will fail if any exception is thrown without the Assert.DoesNotThrow.

but if you want to leave it feel free to do so

**Pull request merged**